src/instrumentsscene/view/systemobjectslayertreeitem.cpp

```diff
@@ -178,6 +178,7 @@ void SystemObjectsLayerTreeItem::onUndoStackChanged(const mu::engraving::ScoreCh
     }
 
     if (shouldUpdateState) {
+        m_systemObjectGroups = collectSystemObjectGroups(m_staff);
```

Comment:
I'm afraid this change will have a negative impact on performance. It scans the entire score for system objects, and we will now call it every time when the user adds / removes / changes a system object. On a small score it doesn't matter as much, but on a larger one it may be more noticeable

Reply:
https://github.com/user-attachments/assets/11b156b7-fafb-44e1-8910-11327053856d

This was the problem I was trying to solve. Dragging a system object layer on another should cancels one of them, which means the remaining one needs to update its elements and state. I've tried a different solution in the last commit, could you tell me if that looks better?

Reply:
Looks better, thank you! However, it can be optimized even further: instead of scanning each staff one by one, we can call collectSystemObjectGroups for all of them (outside of the for loop) and update the state of the items. But it's rather nice to have :)